src/ansible_navigator/utils/version_migration/v1_v2_settings_file.py

```diff
@@ -0,0 +1,384 @@
+"""The v1 to v2 settings file migration."""
+
+
+from typing import Dict
+
+from ...content_defs import ContentView
+from ...content_defs import SerializationFormat
+from ..ansi import COLOR
+from ..ansi import changed
+from ..ansi import info
+from ..dot_paths import MergeBehaviors
+from ..dot_paths import check_path
+from ..dot_paths import get_with_path
+from ..dot_paths import move_to_path
+from ..dot_paths import place_at_path
+from ..serialize import Loader
+from ..serialize import serialize_write_file
+from ..serialize import yaml
+from .definitions import Migration
+from .definitions import MigrationStep
+from .definitions import MigrationType
+
+
+class V1V2SettingsFile(Migration):
+    """The v1 to v2 settings file migration.
+
+    The ordering of the functions in this class is important.
+    It ensures that the migrations are run in the correct order.
+    """
+
+    name = "Version 1 to Version 2 settings file format migration"
+    migration_type: MigrationType = MigrationType.SETTINGS_FILE
+
+    def __init__(self):
+        """Initialize the v1 to v2 settings file migration."""
+        super().__init__()
+        self.content: Dict = {}
+
+    def run(self, *args, **kwargs) -> None:
+        """Perform the v1 to v2 migration.
+
+        :param args: Positional arguments
+        :param kwargs: Keyword arguments
+        """
+        if not self.content:
+            with self.settings_file_path.open("r", encoding="utf-8") as f:
+                try:
+                    self.content = yaml.load(f, Loader=Loader)
+                except (yaml.scanner.ScannerError, yaml.parser.ParserError):
```

Comment:
This should handle JSON file format as well right?

Reply:
Also, can this method be moved to the base class as it looks to handle common logic across migrations?

Reply:
yeah, it'll load the JSON just fine, I did add a test specifically using a JSON file because I wanted to see it work.

We centralized all the writing of files into the serialization code, I'd like to do the same with the loading of yaml too

We do it in 5 other places and this will be 6.  

If it's ok, can we leave this here and I'll log an issue to centralize yaml load into a utils/deserialize file and we can shift them all over